radix-engine/src/model/worktop.rs

```diff
@@ -34,31 +34,29 @@ impl Worktop {
         &mut self,
         amount: Decimal,
         resource_address: ResourceAddress,
-    ) -> Result<Bucket, ResourceContainerError> {
+    ) -> Result<Option<Bucket>, ResourceContainerError> {
```

Comment:
If returning `Option`, should just remove the `Result` part.

Reply:
I respectfully disagree though. `Option` means an optional return, while `Result` represents a fallible response. Not having the resource on worktop doesn't mean an error imo.

Reply:
But there is no possible error in this method?

Reply:
ie this method should be "infallible"

Reply:
There are many errors, e.g. invalid amount/granularity and insufficient balance.

Reply:
I think I missed one point though

Reply:
Should still check if the amount is zero before issuing a `None`.

Reply:
Ahhh I see, didn't notice that `take_by_amount` is just mapped into the return value